### PR TITLE
Explain what the flatten plugin is used for

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -551,14 +549,17 @@
...
         </pluginManagement>
         <plugins>
+            <!--
+                The flatten plugin is required to resolve the "revision" property used in the version tag of POMs in a
+                multi-module setup, see https://maven.apache.org/maven-ci-friendly.html#Install_.2F_Deploy.
+            -->
             <plugin>
                 <groupId>org.codehaus.mojo</groupId>
                 <artifactId>flatten-maven-plugin</artifactId>
                 <version>1.1.0</version>
```
Comment (oheger-bosch) [inline pom.xml]: The most recent version of the flatten-maven-plugin is 1.2.1. It will probably make sense to upgrade the plugin.

File: pom.xml [hunk 2]
```diff
@@ -16,15 +16,13 @@
...
+    <version>${revision}</version>
     <packaging>pom</packaging>
     <name>Antenna</name>
 
     <properties>
         <!-- >This is the main version applied throughout the project. Change this to increase all versions<-->
-        <revision>1.0.0</revision>
-        <!-- >The qualifier can be set to any value and will set the corresponding qualifier in all projects<-->
-        <qualifier>-SNAPSHOT</qualifier>
+        <revision>1.0.0-SNAPSHOT</revision>
```
Comment (neubs-bsi) [inline pom.xml]: could we then not just not use a properties variable at all and instead use project.version?

File: pom.xml (hunk 1 above)
Comment (sschuberth) [reply pom.xml]: Good point, done.

**Comment by oheger-bosch** [comment]
I agree that your changes are in-line with the documentation referenced in the commits. But the comment in the _DependencyTest_ class (and the whole test itself) is still a mystery to me. Do you have any insights here?

**Comment by sschuberth** [comment]
@neubs-bsi FYI, also see [this question](https://stackoverflow.com/q/10582054/1127485) and the accepted answer for a good explanation of the issue.

**Comment by sschuberth** [comment]
> But the comment in the _DependencyTest_ class (and the whole test itself) is still a mystery to me. Do you have any insights here?

Not (yet) really. In a recent call @maxhbr, who added that comment, agreed that it's misleading as "the structure of the repository" as nothing to do with it (it's more about the project being a multi-module project). And maybe the reference to the *flatten* plugin is wrong here, too. Either it should have been the *shade* plugin to begin with, or maybe the whole code is not needed anymore with my recent changes. In fact all of the clean-up work I'm doing has the goal to get rid of this awkward test case and the need to specify transitive dependencies explicitly. We should never need to do that.

**Pull request merged**